src/sql-string.js

```diff
@@ -76,9 +77,54 @@ function escape(val, timeZone, dialect, format) {
       val = val.replace(/\0/g, '\\0');
     }
   } else if (dialect === 'oracle' && typeof val === 'string') {
-    if (val.startsWith('TO_TIMESTAMP') || val.startsWith('TO_DATE')) {
+    if (val.startsWith('TO_TIMESTAMP_TZ') || val.startsWith('TO_DATE')) {
+      const splitVal = val.split(/\(|\)/);
+    
+      if (splitVal.length !== 3 || splitVal[2] !== '') {
+        throw new Error('Invalid SQL function call.');
+      }
+    
+      const functionName = splitVal[0].trim();
+      const insideParens = splitVal[1].trim();
+    
+      if (functionName !== 'TO_TIMESTAMP_TZ' && functionName !== 'TO_DATE') {
+        throw new Error('Invalid SQL function call. Expected TO_TIMESTAMP_TZ or TO_DATE.');
+      }
+    
+      const params = insideParens.split(',');
+    
+      if (params.length < 2) {
+        throw new Error('Invalid format for TO_TIMESTAMP_TZ or TO_DATE');
```

Comment:
Does this mean that we only support the combination of fun + 2 params? Maybe change the error text to something like 

```
`Unexpected input received: ${val}\nSequelize supports TO_TIMESTAMP_TZ or TO_DATE exclusively with a combination of value and format.`
```

Reply:
Error text changed. Users should not receive this error text btw, unless they are doing some pre-processing or are trying to attack